freeipa/src/main/resources/freeipa-salt/salt/freeipa/scripts/freeipa_healthagent_getcerts.sh

```diff
@@ -1,12 +1,49 @@
 #!/usr/bin/env bash
 
+set -x
+set -e
+
 BASE_PATH=/cdp/ipahealthagent
-CERT_FILE=$BASE_PATH/cert.p12
+LOG_FILE=/var/log/freeipa-healthagent-getcert.log
+
+log() {
+    echo $(date) $* >> $LOG_FILE
+}
+
+log "Generate pems for freeipa health agent"
+
+if [ -f "/etc/httpd/alias/cert8.db" ]; then
+  log "Generate privateKey.pem and publicCert.pem by Server-Cert in /etc/httpd/alias/cert8.db"
+  CERT_FILE=$BASE_PATH/cert.p12
+
+  pk12util -o $CERT_FILE -n 'Server-Cert' -d /etc/httpd/alias -k /etc/httpd/alias/pwdfile.txt -W ""
+  openssl pkcs12 -in $CERT_FILE -nocerts -out $BASE_PATH/privateKey.pem -nodes -passout pass: -passin pass:
+  openssl pkcs12 -in $CERT_FILE -clcerts -nokeys -out $BASE_PATH/publicCert.pem -passin pass:
+  rm -f $CERT_FILE
+else
+  log "/etc/httpd/alias/cert8.db is missing"
+
+  PASSWORD_FILE=/var/lib/ipa/passwds/$(hostname)-443-RSA
+  KEY_FILE=/var/lib/ipa/private/httpd.key
+
+  if [ ! -f "$KEY_FILE" ]; then
+    log "$KEY_FILE is missing, unable to generate privateKey.pem"
+  elif [ ! -f "$PASSWORD_FILE" ]; then
+    log "$PASSWORD_FILE is missing, unable to generate privateKey.pem"
+  else
+    log "Generate privateKey.pem by $KEY_FILE and $PASSWORD_FILE"
+    openssl rsa -in $KEY_FILE -passin file:$PASSWORD_FILE -text > $BASE_PATH/privateKey.pem
+  fi
+
+  CRT_FILE=/var/lib/ipa/certs/httpd.crt
+  if [ ! -f "$CRT_FILE" ]; then
+    log "$CRT_FILE is missing, unable to generate publicCert.pem"
```

Comment:
I think this might should cause a failure too

Reply:
It will, please check my comment above around PASSWORD_FILE